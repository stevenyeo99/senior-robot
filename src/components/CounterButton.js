import { Component } from "react";

class CounterButton extends Component {

    constructor() {
        super();
        this.state = {
            count: 0
        }
    }

    updateCount = () => {
        console.log('updateCount');
        this.setState({
            count: this.state.count + 1
        });
    }

    render() {
        console.log('CounterButton Render', this.state.count);
        return (
            <button id='button' color={this.props.color} onClick={this.updateCount}>Count: {this.state.count}</button>
        )
    }
}

export default CounterButton;