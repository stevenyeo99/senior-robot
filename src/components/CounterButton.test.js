import { shallow } from 'enzyme';

import CounterButton from './CounterButton';

it('test counter button component', () => {
    expect(shallow(<CounterButton color='red' />)).toMatchSnapshot();
});

it('test counter button handler', () => {
    const button = shallow(<CounterButton color='green' />);
    button.find('#button').simulate('click');
    button.find('#button').simulate('click');
    expect(button.state()).toEqual({count: 2});
    console.log(button.props());
});