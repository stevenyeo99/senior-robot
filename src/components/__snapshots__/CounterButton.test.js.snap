// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test counter button component 1`] = `
<button
  color="red"
  id="button"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
